File: 3702-Longest-Subsequence-With-Non-Zero-Bitwise-XOR/Solution.swift
class Solution {
    func longestSubsequence(_ nums: [Int]) -> Int {
        let n = nums.count
        var xor = 0
        var hasNonZero = false

        for num in nums {
            xor ^= num
            if num != 0 {
                hasNonZero = true
            }
        }

        if xor != 0 {
            return n
        }
        return hasNonZero ? n - 1 : 0
    }
}

File: 3702-Longest-Subsequence-With-Non-Zero-Bitwise-XOR/Solution.kt
class Solution {
    fun longestSubsequence(nums: IntArray): Int {
        val n = nums.size
        var xor = 0
        var hasNonZero = false

        for (num in nums) {
            xor = xor xor num
            if (num != 0) {
                hasNonZero = true
            }
        }

        return if (xor != 0) n else if (hasNonZero) n - 1 else 0
    }
}
